hpx/parallel/algorithms/detail/predicates.hpp

```diff
@@ -116,7 +155,11 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1) { namespace detail
         HPX_HOST_DEVICE
         HPX_FORCEINLINE static Iter call(Iter iter, Stride offset)
         {
+#if defined(HPX_HAVE_CUDA) && defined(__CUDACC__)
+            hpx::detail::advance(iter, offset);
+#else
             std::advance(iter, offset);
+#endif
```

Comment:
Can't we get rid of the `std::advance` altogether? Your (new) implementation should do just fine, shouldn't it?